pkg/apis/monitoring/v1beta1/alertmanager_config_types.go

```diff
@@ -863,6 +865,23 @@ type TelegramConfig struct {
 	HTTPConfig *HTTPConfig `json:"httpConfig,omitempty"`
 }
 
+type MSTeamsConfig struct {
+	// Whether to notify about resolved alerts.
+	// +optional
+	SendResolved *bool `json:"sendResolved,omitempty"`
+	// MSTeams webhook url
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern:="^http(s)?://.+$"
+	WebhookUrl string `json:"webhookUrl"`
+	// +kubebuilder:validation:Required
```

Comment:
Hmm I think that this field and the next one `Text` can be kept optional because Alertmanager will use a default if unset.